### PR TITLE
feat: supporting inline folding

### PR DESCRIPTION
**Pull request by kelly-thai**
## Summary

Implemented inline folding within studio. Package paths will now fold by default upon entering text mode, and will unfold/refold when users double click in front of an element. Also included a button that will fold/unfold all paths. These changes will allow for better code readability and enhance user experience by cleaning up the code editor.

## How did you test this change?

- [ ] Test(s) added
- [x] Manual testing (please provide screenshots/recordings)
- [ ] No testing (please provide an explanation)

Manually tested across different projects that all paths are folded upon entering text mode and that behavior when user is making edits is expected. Visually inspected paths throughout a project, and scanned for edge cases. Manually tested different scenarios in terms of folding/unfolding, editing different parts of the code, and using the button in different situations.

[Demo Video](https://github.com/finos/legend-studio/assets/73408381/d2f4f0b1-a0a0-4cf9-bdcc-254b18a6ac53)

### CODE REVIEW

**Comment by YannanGao-gs** [comment]
push on behalf of @kelly-thai

**Comment by YannanGao-gs** [comment]
demo video: 

https://github.com/finos/legend-studio/assets/73408381/d2f4f0b1-a0a0-4cf9-bdcc-254b18a6ac53

**Comment by YannanGao-gs** [comment]
demo video: 
https://github.com/finos/legend-studio/assets/73408381/1f066684-15a2-4fbf-af51-e1190d09e71c

**Comment by YannanGao-gs** [comment]
https://github.com/finos/legend-studio/assets/73408381/d2f4f0b1-a0a0-4cf9-bdcc-254b18a6ac53

File: packages/legend-application-studio/src/components/editor/editor-group/GrammarTextEditor.tsx [hunk 1]
```diff
@@ -974,6 +987,200 @@ export const GrammarTextEditor = observer(() => {
       );
   }
 
+  //clear all inline folding decorations
```
Comment (akphi) [inline packages/legend-application-studio/src/components/editor/editor-group/GrammarTextEditor.tsx:990]: all of these helper functions, if possible, should be declared outside of the render method

File: packages/legend-application-studio/src/components/editor/editor-group/GrammarTextEditor.tsx [hunk 2]
```diff
@@ -974,6 +987,200 @@ export const GrammarTextEditor = observer(() => {
...
+  //find paths to fold and initialize decorations
+  useEffect(() => {
+    if (editor) {
+      clearDecorations();
+      makeDecorations();
+    }
+  }, [editor, makeDecorations]);
+
+  //track user clicks; check for double clicks
+  useEffect(() => {
```
Comment (akphi) [inline packages/legend-application-studio/src/components/editor/editor-group/GrammarTextEditor.tsx:1152]: do not do this, instead, use `monaco-editor` event hooks

**Pull request closed without merging**